llvm/test/MC/RISCV/fixups-expr.s

```diff
@@ -47,3 +51,28 @@ G2:
 # RELAX: 0x1C R_RISCV_SUB8 .L1 0x0
 # RELAX: 0x1D R_RISCV_ADD8 G2 0x0
 # RELAX: 0x1D R_RISCV_SUB8 G1 0x0
+
+.dword .L3-.L2
```

Comment:
Seems LoongArch behaves different from RISC-V on this case. RISC-V emits ADD/SUB relocs while LoongArch not.

Reply:
LoongArch not emit relocs because it doesn't collect relocations in `handleAddSubRelocations`. It returned because relax is disabled and the A and B are both in the same section, that can calculate out the absolute diff in asm stage.

Reply:
But the diff is not fixed because there are relaxable instructions (`call extern`) before these 2 labels.

On LoongArch, for example:
```
.text
la.pcrel $t0, foo
L1:
.align 4
L2:
nop
```
When relax is enable, L1 is 4 or 8, L2 is 16. L2-L1=12 or 8.

Reply:
So check `shouldInsertExtraNopBytesForCodeAlign` before add `Displacement`. We'll do something like RISCV to handle alignment directive, but now it miss the relax alignment part and `shouldInsertExtraNopBytesForCodeAlign` return false always.

Reply:
Oh, I see. Thanks.